discord/commands/options.py

```diff
@@ -390,6 +387,48 @@ def to_dict(self) -> dict:
     def __repr__(self):
         return f"<discord.commands.{self.__class__.__name__} name={self.name}>"
 
+    @property
+    def autocomplete(self):
+        """
+        The autocomplete handler for the option. Accepts a callable (sync or async)
+        that takes a single required argument of :class:`AutocompleteContext`.
+        The callable must return an iterable of :class:`str` or :class:`OptionChoice`.
+        Alternatively, :func:`discord.utils.basic_autocomplete` may be used in place of the callable.
+
+        Returns
+        -------
```

Comment:
Maybe instead of adding the ``Returns`` section do something like: ``AutocompleteFunction = Union[...]`` on top of the file. This would also allow to type annotate the ``value`` parameter on the autocomplete setter.

Reply:
Yeah good idea